### PR TITLE
Fixing for docker-machine on OS-X/Windows

### PR DESCRIPTION
**Pull request by inadarei**
For people who run Docker in Docker Machine, the IP is not going to be 127.0.0.1, but something more like 10.211.55.5. Hardcoded 127.0.0.1 in the instructions is both unnecessary and actually breaks things for Mac and Windows users.

### CODE REVIEW

**Comment by envygeeks** [comment]
I'm confused as to your logic.  127.0.0.1 is the IP address it's locally allocated to for you (or the world) to access, and it's very necessary otherwise we are exposing users instances of which I am never going to accept by default.

That has (or should not have) nothing to do with the IP that Docker machine uses.  Unless Docker machine confuses that logic somehow, and that's not our responsibility (and I don't know that I'm willing to accept that responsibility because...) that's a surface on boot2docker or docker-machine.

Elaborate please.

**Comment by envygeeks** [comment]
Let me elaborate further:

* OS X does accept connections to 127.0.0.1
* Windows does accept connections to 127.0.0.1
* Linux homes on 127.0.0.1 inside of it's stack.

If I do `-p 127.0.0.1:4000:4000` then the listener for "4000" will end up on `127.0.0.1` instead of `0.0.0.0` < What we are helping the user guard against.  Your change makes it so that we listen on every interface which means that we are exposing the users instance to the entire internet if they have an improperly configured firewall.  Docker Machine has nothing to do with that since that's a system thing, that goes through *your interfaces* not theirs. 

If it doesn't work that way I think it might be a misconfigured firewall on your behalf or a problem with boot2docker or docker-machine (though I have no problems with it on my Windows machine as I just tried it... and it works as it stands.)

Unless something changed that I am not aware of (and it hasn't changed in upstream Docker as I just tested it on my personal computer -- which is Linux and it still behaves exactly as expected.) So I don't know where this is coming from.

**Comment by inadarei** [comment]
Sure, let me explain. 

Current standard for running Docker on OS-X is: using [Docker Machine](https://www.docker.com/docker-toolbox). What that does basically is: starts a Linux VM (VirtualBox by default, but you can use [Parallels](http://kb.parallels.com/en/123356) or [VMWare Fusion](http://blog.javabien.net/2015/08/20/better-docker-on-osx-with-docker-machine-boot2docker-and-vmware/) as well)

Docker machine runs Docker client on OS-X itself, whereas docker server runs in the Linux VM it creates. This is where things become interesting. When you run that docker command the process launches inside the Linux VM. If you log into the Linux VM the IP is 127.0.0.1, sure, but from the perspective of OS X: it absolutely is not. The IP of the virtual machine is going to be some 10.x.x.x IP. 

The problem with your instruction is that: when anybody opens Chrome or Safari on their Macs, if they type `127.0.0.1:4000` that is not going to the Virtual Machine and is not going Jekyll - it's going nowhere. If they type 10.211.55.5 (that's the local IP for me) - they still cannot access anything, because they ARE hitting the virtual machine but you just told the Docker Host to ignore requests on any IP but 127.0.0.1 so it doesn't return anything.

Basically: your instructions don't work due to how networking is set up with Docker Machine. And it's not Docker Machine's fault. It's how things work.

OS-X shuts down all ports by default. Port 4000 is never open to the world by default on OS-X, so there is really no security issue whatsoever here.

**Comment by inadarei** [comment]
I recorded a screencast to show the problem: http://f.cl.ly/items/0K3K3h0a3Q243u3e3Y0D/jekyll-docker-networking.mp4

**Comment by inadarei** [comment]
Last but not least: running docker without IP restriction leads to the exact same exposure that everybody would get if they ran Jekyll natively (without Docker). If it were safe for native, it should be safe for Docker-ized version as well.

**Comment by envygeeks** [comment]
On your last comment, that's a negative, Jekyll by default attaches to localhost, we are the ones who enforce "0.0.0.0" for *you* because without it Docker wouldn't work at all: https://github.com/jekyll/docker/blob/master/copy/usr/local/bin/jekyll#L18 actually we do quite a lot on your behalf to make sure that every scenario I can think of is as painless as possible and you can get up and running quickly without effort.

**Comment by inadarei** [comment]
If you are really uncomfortable removing IP restriction, then I would suggest adding a comment right below it, which explains that when using Docker Machine, the IP will most likely not be 127.0.0.1 and people should instead use the IP address returned by the following command:

```
docker-machine ip `docker-machine active`
```

**Comment by inadarei** [comment]
The fully automated command for anybody using Docker Machine would be:

```
docker run --rm -v $(pwd):/srv/jekyll -it -p $(docker-machine ip `docker-machine active`):4000:4000 jekyll/jekyll jekyll s
```

For people not using Docker Machine (e.g. on Linux) it would be what you had originally.

**Comment by envygeeks** [comment]
I haven't decided yet, because everything you said I already know, and I know that's *not* how Docker networking works in a native environment where it forwards traffic and proxies it, and it could clearly do the same for Docker machine and Boot2Docker for OS X... Windows I don't know... I know it supports forwarding but compared to OS X and especially Linux it's not easy.

What I am saying is, nothing listen's on "127.0.0.1" on native Docker, it's forwarded and there is a proxy that handles all that because the concept is *exactly* the same (I will admit I haven't dug into their new networking stack to see if it's changed things and it clearly might have)

**Comment by envygeeks** [comment]
I should rephrase that before people take it wrong, nothing in *your  app* listens on 127.0.0.1, the proxy listens on 127.0.0.1 and traffic gets fowarded and that same high level concept could apply to docker-machine.

**Comment by inadarei** [comment]
Thank you, Jordon, for a quick fix.
